Saving and restoring scroll position in composition mode (#3268)
microsoft/BotFramework-WebChat: docs/HOOKS.md

When called, this hook will return an array of notifications.

## `useObserveScrollPosition`

<!-- prettier-ignore-start -->
```js
useObserveScrollPosition(observer: (ScrollObserver? | false), deps: any[]): void

type ScrollObserver = (position: ScrollPosition) => void;

type ScrollPosition {
  scrollTop: number
}
```
<!-- prettier-ignore-end -->

This function accept an observer function. When the scroll position has changed, the observer function will be called with the latest `ScrollPosition`.

The `position` argument can be passed to [`useScrollTo`](#usescrollto) hook to restore scroll position.

Since the observer function will be called rapidly, please keep the code in the function as lightweight as possible.

To stop observing scroll positions, pass a falsy value to the `observer` argument.

> > If there are more than one transcripts, scrolling any of them will trigger the observer function, and there is no clear distinction of which transcript is being scrolled.

Reviewer: ```suggestion
> If there is more than one transcripts, scrolling any of them will trigger the observer function, and there is no clear distinction of which transcript is being scrolled.
```

-  `typing` lists participants who did not explicitly stopped typing. This list is a superset of `activeTyping`.
-  `visible` indicates whether typing indicator should be shown in normal case. This is based on participants in `activeTyping` and their `role` (role not equal to `"user"`).

## `useScrollTo`

<!-- prettier-ignore-start -->
```js
useScrollTo(): (position: ScrollPosition, options: ScrollOptions) => void

type ScrollOptions {
  behavior: 'auto' | 'smooth'
}

type ScrollPosition {
  scrollTop: number
}
```
<!-- prettier-ignore-end -->

> This function will return a function that, when called, will scroll the transcript to the specific scroll position.

Reviewer: ```suggestion
This hook will return a function that, when called, will scroll the transcript to the specific scroll position.
```

Author: Looks like we should also update other hooks description. I will file a doc issue.

```
<!-- prettier-ignore-end -->

> This function will return a function that, when called, will scroll the transcript view to the end using smooth scrolling.

Reviewer: ```suggestion
This function will return a function that, when called, will smoothly scroll the transcript view to the end.
```
